src/Service/Services/ResolverMiddleware.cs

```diff
@@ -200,8 +201,9 @@ private static object PreParseLeaf(IMiddlewareContext context, string leafJson)
             {
                 ByteType => byte.Parse(leafJson),
                 SingleType => Single.Parse(leafJson),
-                DateTimeType => DateTimeOffset.Parse(leafJson),
+                DateTimeType => DateTimeOffset.Parse(leafJson, DateTimeFormatInfo.InvariantInfo, DateTimeStyles.AssumeUniversal),
```

Comment:
Your PR description notes:
> In GraphQL, while parsing a values of datetime, datetime2, they should be treated as UTC, otherwise HotChocolate tries to convert into UTC and max values stored in the db then cause overflow exceptions.

The first part i understand: Any datetime input provided by the user is assumed to be in UTC, which means an offset is required if they want to refer to their own local time.

```csharp
string dateString;
DateTimeOffset offsetDate;

dateString = "05/01/2008 6:00:00";

// Assume time is UTC
offsetDate = DateTimeOffset.Parse(dateString, null, DateTimeStyles.AssumeUniversal);
Console.WriteLine(offsetDate.ToString());   // Displays 5/1/2008 6:00:00 AM +00:00
```

So when `DateTimeOffset.Parse` is executed here, an offset +00:00 is then added [docs](https://learn.microsoft.com/en-us/dotnet/api/system.globalization.datetimestyles?view=net-6.0#fields)
> If no time zone is specified in the parsed string, the string is assumed to denote a UTC. 

I'm not following the last part:
> HotChocolate tries to convert into UTC and max values stored in the db then cause overflow exceptions.

Reply:
**updated Description**. By using  `DateTimeStyles.AssumeUniversal` , wecan parse DateTime strings without requiring an explicit time zone indicator, as long as the input is in the expected UTC format. It doesn't try to convert it, it just assumes the time provided is already in the UTC format.